homeworks/5/src/Fetch.jsx

```diff
@@ -0,0 +1,30 @@
+import React, { useState, useEffect } from "react";
+
+const getPokemonUrl = (id) => `https://pokeapi.co/api/v2/pokemon/${id}/`
+
+export default function Fetcher() {
+	const [id, setId] = useState(1)
+	const [pokemons, setPokemons] = useState([])
+
+	useEffect(() => {
+		let mounted = true
+		if (mounted) {
+			fetch(getPokemonUrl(id))
+				.then(response => response.json()
+					.then(json => setPokemons([...pokemons, json.name])
```

Comment:
```suggestion
					.then(json => setPokemons(pokes => [...pokes, json.name])
```
U teoriji ti ovaj useEffect ovisi o pokemons varijabli i može doći do potencijalnih prepisivanja (ne vjerujem da bi u praksi, ali u teoriji da), ovak ne ovisi, state setter na ovaj način ima pristup trenutnoj vrijednosti statea